File: 70.climbing-stairs.py
#
# @lc app=leetcode id=70 lang=python3
#
# [70] Climbing Stairs
#

# @lc code=start
class Solution:
    def climbStairs(self, n: int) -> int:
        fib = []
        for i in range(n):
            if i is 0:
                fib.append(1)
            elif i is 1:
                fib.append(2)
            else:
                fib.append(fib[i-2]+fib[i-1])
        
        return fib[n-1]

# @lc code=end

File: 70-climbing-stairs/70-climbing-stairs.cpp
class Solution {
public:
    int climbStairs(int n) {
        int dp[n];
        
        for(int i = 0; i < n; i++){
            if(i == 0)
                dp[i]=1;
            else if(i==1)
                dp[i]=2;
            else
                dp[i]= dp[i-2]+dp[i-1];
        }
        
        return dp[n-1];
    }
};